src_c/time.c

```diff
@@ -191,68 +255,41 @@ time_wait(PyObject *self, PyObject *arg)
     return PyInt_FromLong(SDL_GetTicks() - start);
 }
 
-#if IS_SDLv2
 static PyObject *
-time_set_timer(PyObject *self, PyObject *arg)
+time_set_timer(PyObject *self, PyObject *args, PyObject *kwargs)
 {
-    SDL_TimerID newtimer;
-    int ticks = 0;
-    int once = 0;
-    SDL_EventType event;
-    size_t index;
-    if (!PyArg_ParseTuple(arg, "ii|i", &event, &ticks, &once))
+    int ticks, once = 0, repeat = 0;
+    PyObject *obj;
+    pgEventObject *e;
+
+    static char *kwids[] = {
+        "event",
+        "millis",
+        "once",
+        "repeat",
```

Comment:
I wonder if repeat is the best name?

Reply:
repeat is fine IMHO.
Another alternative would be numevents

Reply:
We need a unit test for this repeat arg.

Reply:
Let's go with repeat, unless someone objects... what do you think @MyreMylar @Starbuck5 ?

Reply:
I already added a repeat unit test :)

Reply:
@robertpfeiffer what do think of the name for the argument

Reply:
I don't mind repeat. I think I would assume from the name that it was a boolean flag to make the event timer continuously repeat infinitely.

Reply:
Hmm valid point. Do you have any other names in mind

Reply:
I agree with MyreMalar - repeat sounds like a flag. I'm partial to illume's proposed `num_times` in the issue. I like that or just plain `times`.

Reply:
We could call it `loops` since `mixer` and `mixer.music` uses it in `.play()`.

Reply:
We can discuss this further in #2347